diff --git a/node_modules/@mgcrea/pino-pretty-compact/lib/utils/chalk.js b/node_modules/@mgcrea/pino-pretty-compact/lib/utils/chalk.js
index b461467..ea35b18 100644
--- a/node_modules/@mgcrea/pino-pretty-compact/lib/utils/chalk.js
+++ b/node_modules/@mgcrea/pino-pretty-compact/lib/utils/chalk.js
@@ -9,7 +9,20 @@ const chalkString = (s) => chalk_1.default.green(`'${s}'`);
 exports.chalkString = chalkString;
 const chalkKeyword = (s) => chalk_1.default.blue(`${s}`);
 exports.chalkKeyword = chalkKeyword;
-const chalkJson = (s) => chalk_1.default.grey(JSON.stringify(s));
+//Makes it not choke on circular objects
+const getCircularReplacer = () => {
+    const seen = new WeakSet();
+    return (key, value) => {
+        if (typeof value === "object" && value !== null) {
+            if (seen.has(value)) {
+                return;
+            }
+            seen.add(value);
+        }
+        return value;
+    };
+};
+const chalkJson = (s) => chalk_1.default.grey(JSON.stringify(s, getCircularReplacer()));
 exports.chalkJson = chalkJson;
 const chalkStringArray = (a) => `[ ${a.map(exports.chalkString).join(', ')} ]`;
 exports.chalkStringArray = chalkStringArray;
