diff --git a/node_modules/axios/index.d.ts b/node_modules/axios/index.d.ts
index 0b9e1d3..60ac71b 100644
--- a/node_modules/axios/index.d.ts
+++ b/node_modules/axios/index.d.ts
@@ -247,6 +247,8 @@ export interface AxiosStatic extends AxiosInstance {
   all<T>(values: Array<T | Promise<T>>): Promise<T[]>;
   spread<T, R>(callback: (...args: T[]) => R): (array: T[]) => R;
   isAxiosError(payload: any): payload is AxiosError;
+  //Fixes incorrect typings
+  AxiosError: typeof AxiosError
 }
 
 declare const axios: AxiosStatic;
